In [1]:
#Imports
import pandas as pd
import matplotlib.pyplot as plt
from scipy.stats import linregress
%matplotlib inline
from pathlib import Path
import seaborn as sns
  

In [2]:
# CSV files:
file_path = Path("Resources/epa-sea-level.csv")

# Create DataFrames
df = pd.read_csv(file_path)
df.head()

Year  CSIRO Adjusted Sea Level  Lower Error Bound  Upper Error Bound
0  1880                  0.000000          -0.952756           0.952756
1  1881                  0.220472          -0.732283           1.173228
2  1882                 -0.440945          -1.346457           0.464567
3  1883                 -0.232283          -1.129921           0.665354
4  1884                  0.590551          -0.283465           1.464567

In [1]:
#Scatter plot of Dataset
df.plot.scatter(x='Year', y='CSIRO Adjusted Sea Level')


In [ ]:
# sns.set_style('white')
sns.set_style('ticks')
sns.regplot(x='Year', y='CSIRO Adjusted Sea Level',color='g',data=df);

In [ ]:
# Create first line of best fit
x=df['Year'], 
y=df['CSIRO Adjusted Sea Level']
res = linregress(x,y)

x_pred = pd.Series([i for i in range (1880,2051)])
y_pred = res.slope*x_pred + res.intercept
print(res)


In [ ]:
#Plot modeling predictions
plt.plot(x_pred, y_pred,color='r')
plt.xlabel('Year')
plt.ylabel('CSIRO Adjusted Sea Level')
plt.title('Sea Level Predictions to 2050' );


In [ ]:
# Create second line of best fit
new_df = df.loc[df['Year']>=2000]
new_df.head()


In [ ]:
sns.set_style('ticks')
sns.regplot(x='Year', y='CSIRO Adjusted Sea Level',color='b',data=new_df);

In [ ]:
# Sea level increase from 2000 to 2050
new_x = new_df['Year']
new_y = new_df['CSIRO Adjusted Sea Level']
res_2 = linregress(new_x, new_y)
x_pred2 = pd.Series([i for i in range(2000,2051)])
y_pred2 = res_2.slope*x_pred2 + res_2.intercept
plt.xlabel('Year')
plt.ylabel('CSIRO Adjusted Sea Level')
plt.title('Rapid Increase in Water Levels beyond 2000')
plt.plot(x_pred2, y_pred2, 'green');

In [ ]:
# Increase in sea level occuring faster than anticipated beyond 2000.
plt.plot(x_pred, y_pred,x_pred2, y_pred2);
plt.xlabel('Year')
plt.ylabel('CSIRO Adjusted Sea Level')
plt.title('Sea Levels Increases more than predicted from 2000')
#plt.figsize(30,30)

The LinRegressor sea level predictor model shows an expected 26-77 cm in sea levels to 2050 as the planet warms to 1.5 degrees celcius above pre-industrial levels. Should these levels be realized, it is expected that more than 1/3 of the earth's land mass will be underwater by 2050. The warming of the earth not only increases sea levels but also reduces the sea ice levels significantly as previously mention and the important role that the sea ice plays in regulating the planet's climate by reflecting sunlight and heat back to space through a process called .... The ice-albedo feedback which is a positive feedback climate process where a change in the area of ice caps, glaciers and sea ice alters the albedo and surface temperature of the planet.

The model predicts that this trend will continue for the forseable future. Simple put, when the ice melts, the sea level rises. As seas warm and ice melts, the earth's oceans have risen steadily. The second slope above shows an even more alarming statistic which says, sea level rise is accelerating much more than initially predicted. to put it simply, for the past 25 years, sea level is not only rising, but the rate of that rise is getting faster. As we saw from the heat map, increased ice melting in Greenland and Antactica is primarily responsible for the acceleration. The the predictions are realised and this accelration continues, global sea levels could be 26 inches higher by 2100.